polyply/src/load_library.py

```diff
@@ -13,75 +13,112 @@
 # limitations under the License.
 
 import os
-import pathlib
+from pathlib import Path
 import vermouth
 from vermouth.gmx.rtp import read_rtp
 from vermouth.citation_parser import read_bib
+from vermouth.log_helpers import StyleAdapter, get_logger
 from polyply import DATA_PATH
 from .ff_parser_sub import read_ff
+from .build_file_parser import read_build_file
 from .polyply_parser import read_polyply
 
+LOGGER = StyleAdapter(get_logger(__name__))
+BUILD_FILE_PARSERS = {'bld': read_build_file}
 FORCE_FIELD_PARSERS = {'rtp': read_rtp, 'ff': read_ff, 'itp': read_polyply, 'bib': read_bib}
 
+def get_parser(file_, file_parsers, is_lib_file):
+    """
+    check if file can be parsed and
+    if possible return the respective parser
+
+    Parameters
+    ----------
+    file_: class:`pathlib.PosixPath`
+        path to file
+    file_parsers: dict
+        dictionary of available file parsers
+    is_lib_file: bool
+        indicates whether the provided path is from a data library
+    """
+    file_extension = file_.suffix[1:]
+    if file_extension in file_parsers:
+        return file_parsers[file_extension]
+    elif not is_lib_file:
+        msg = "Cannot parse user provided file with extension {file_extension}."
+        raise IOError(msg)
```

Comment:
It might be useful to add a third exception in case an unkown file is added to the library